src/commands/CommandHandler.ts

```diff
@@ -38,90 +40,93 @@ import { execShutdownRoomCommand } from "./ShutdownRoomCommand";
 import { execAddAliasCommand, execMoveAliasCommand, execRemoveAliasCommand, execResolveCommand } from "./AliasCommands";
 import { execKickCommand } from "./KickCommand";
 import { execMakeRoomAdminCommand } from "./MakeRoomAdminCommand";
-import { parse as tokenize } from "shell-quote";
 import { execSinceCommand } from "./SinceCommand";
-
+import { Lexer } from "./Lexer";
 
 export const COMMAND_PREFIX = "!mjolnir";
 
 export async function handleCommand(roomId: string, event: { content: { body: string } }, mjolnir: Mjolnir) {
-    const cmd = event['content']['body'];
-    const parts = cmd.trim().split(' ').filter(p => p.trim().length > 0);
+    const line = event['content']['body'];
+    const parts = line.trim().split(' ').filter(p => p.trim().length > 0);
 
-    // A shell-style parser that can parse `"a b c"` (with quotes) as a single argument.
-    // We do **not** want to parse `#` as a comment start, though.
-    const tokens = tokenize(cmd.replace("#", "\\#")).slice(/* get rid of ["!mjolnir", command] */ 2);
+    const lexer = new Lexer(line);
+    lexer.token("command"); // Consume `!mjolnir`.
```

Comment:
what happens when the lexer fails then? how different is it from what happens now?